src/logger.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import time
+import logging
+import datetime
+from enum import Enum
+from abc import ABC
+
+
+class BaseFormatter(ABC, logging.Formatter):
+
+    def formatTime(self, record):
+        """
+        Return the creation time of the specified LogRecord as formatted text.
+        This method should be called from format() by a formatter which
+        wants to make use of a formatted time.
+        It provides an ISO 8601 format with milliseconds.
+        """
+        ct = self.converter(record.created)
+        ct_with_ms = time.mktime(ct) + (record.msecs / 1000)
+        ct_full = datetime.datetime.fromtimestamp(ct_with_ms)
```

Comment:
Could we add the timezone info here? That would reduce the confusion when trying to correlate different log events. 
 Is that info available in the record? Or is the timestamp always in the system timezone? In that case we could use something link:

```python
ct_full = datetime.datetime.fromtimestamp(ct_with_ms, datetime.datetime.utcnow().astimezone().tzinfo)
```

Reply:
@schans I just made a change to expose a `LOG_TZ` variable to choose time formatted in local timezone or UTC timezone. 

The `time` fields is formatted in ISO 8601 format (e.g `2022-05-28T14:24:04.037058+00:00`)

Reply:
Thanks.

From my local tests and what I understand from the docs, the isoformat() function will only add the TZ info if that is present in the timestamp (the +00:00 part), but maybe that's something local to my setup.

Reply:
I'm using the record `created` (POSIX timestamp) and format it like that:
```
datetime.fromtimestamp(record.created, log_tz).isoformat()
```
So I format to IS0 8601 format a `datetime` with `tzinfo`.

From my tests it gives:

**UTC TZ**
```
{"time": "2022-05-29T11:22:11.377220+00:00", "level": "INFO", "msg": "Starting collector"}
```


**LOCAL TZ**
```
{"time": "2022-05-29T07:21:56.590056-04:00", "level": "INFO", "msg": "Starting collector"}
```

Reply:
Nice! Perfect.